token/inc/token.h

```diff
@@ -164,10 +168,27 @@ typedef enum Token_TokenInstruction_Tag {
      *
      *   * Multisignature owner/delegate
      *   0. `[writable]` The account to burn from.
-     *   1. `[]` The account's multisignature owner/delegate
+     *   1. `[]` The account's multisignature owner/delegate.
      *   2. ..2+M '[signer]' M signer accounts.
      */
     Burn,
+    /**
+     * Burns all the tokens in the account and transfers all its SOL to the destination account.
+     *
+     * Accounts expected by this instruction:
+     *
+     *   * Single owner/delegate
+     *   0. `[writable]` The account to burn.
+     *   1. '[writable]' The destination account.
+     *   2. `[signer]` The account's owner.
+     *
+     *   * Multisignature owner/delegate
+     *   0. `[writable]` The account to burn.
+     *   1. '[writable]' The destination account.
+     *   2. `[]` The account's multisignature owner.
+     *   3. ..3+M '[signer]' M signer accounts.
+     */
+    BurnAccount,
```

Comment:
Call this Unwrap or Release feels better.   Burn is a scary thing, I never want to burn my native SOL

Reply:
👻

Reply:
How does `Withdraw` sound?

Reply:
🤔, not my fav.  A normal `Transfer` could be considered a withdrawal.
 
Some ideas in order of preference:
1.0 `UnwrapNativeToken`
1.0 `UnwrapNative`
2. `ReleaseNative`
3. `ReleaseAccount`

Reply:
`BurnAccount` supports both native and non-native and in effect is burning the tokens and withdrawing the lamports and in the end its the Account that is burned since it will be purged.  not very excited about `ReleaseAccount`,  The word "Release" sounds too vague and it doesn't correlate to any other verbiage in the instructions.  If we have "Capture" then maybe "Release" ;-) Could be very milktoast and call it `UninitializeAccount`

Reply:
`RepossesAccount`?

Reply:
wrap/unwrap is used in eth.  Another synonym: box/unbox.

For non-native tokens, we don't really need `BurnAccount` to actually burn, there's already an instruction for this.
So we could require that a non-native token have a zero balance to keep the intent clear.

Reply:
Not excited about adding another instruction for wrap/box.  `UnitializeAccount` is a good complement to `InitializeAccount` .  Non-native accounts are initialized with zero balance and uninitialized only if zero balance and native accounts are initialized to the Lamport balance and uninitialized at the Lamport balance.  Sticky point is that the account is left with zero balance and will be purged, not just left in an uninitialized state.  Could call it `DestroyAccount` with the same behavior described above

Reply:
`DestroyAccount` I'm a little bearish on, it's not really the token program that does the destroying (deallocation).  The runtime does this in response to the `UnitializeAccount`.   Maybe time for some run in #terminology to see if others have better name ideas?

Reply:
Fwiw, I like `ReleaseAccount` best of those options so far. That suggests to me that my SOL are coming back to me, but that the tokens (if non-native) will disappear.

Reply:
`RedeemAccount`, `CloseAccount`, `DischargeAccount` come to mind. None a particularly good antonym to `InitializeAccount`.  How married to "Initialize" are we?